app/Http/Controllers/Maps/MapGroupController.php

```diff
@@ -5,12 +5,41 @@
 use App\Facades\CampaignLocalization;
 use App\Http\Controllers\Controller;
 use App\Http\Requests\StoreMapGroup;
+use App\Facades\Datagrid;
+use App\Http\Requests\ReorderGroups;
+use App\Datagrids\Actions\GroupDatagridActions;
 use App\Models\Campaign;
 use App\Models\Map;
 use App\Models\MapGroup;
 
 class MapGroupController extends Controller
 {
+    /** @var string|null The datagrid controlling the bulk actions*/
+    protected $datagridActions = GroupDatagridActions::class;
+
+    /**
+     * Index
+     */
+    public function groups(Map $map)
+    {
+        $campaign = CampaignLocalization::getCampaign();
```

Comment:
I can call this page in an incognito window. Only a user who can edit the map should have access to it.